### DISCUSSION TITLE
What are the recommended steps for resolving Sunbird KP issues that might be unrelated to adopters?

### DISCUSSION THREAD

**Discussion by joffinjoy** [Sunbird-inQuiry/Community#20] [Q&A]
Currently we are having some issues with question/questionset publish API. Basically the apis are returning success response, but the status of the question/questionset remains as 'review'. This is happening on the dock-dev VDN portal too. 

I can confirm that it was working fine until last Thursday and we haven't worked on KP since then. As such we believe this is an issue elsewhere. So what are the recommended steps to resolve issues like these? Should we be reporting this matter in some specific channel or to any specific person?

This particular issue has been a blocker for multiple teams of ours. We would like to know what can be done in future for issues like these.

Response from publish API:
```
{
    "id": "api.questionset.publish",
    "ver": "3.0",
    "ts": "2022-07-13T06:39:48ZZ",
    "params": {
        "resmsgid": "6d5475f6-1ccb-44e3-b194-3b58bd42cdc3",
        "msgid": null,
        "err": null,
        "status": "successful",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "message": "QuestionSet is successfully sent for Publish",
        "identifier": "do_11357930337583923214926"
    }
}
```
cc: @aks30 @vijiurs

**Comment by krgauraw** [comment]
@joffinjoy publishing an object is an asynchronous process  using  sunbird. The Process involves a publish api and a respective backend job.  When publish api returns 200 response means it submitted a publish request to the backend job for further processing.

In your case, publish api returning success response means the service is working fine but if the object status remains same as before invoking publish api,  it indicates that  some problem exist with backend job pipeline (in this case questionset-publish flink job). 

If you get into such situation, you can try below steps:
1. Login to Rancher / Kubernetes Server to check if the job is working fine.
2. If the job is running, check the logs if the previous publish event stuck because of some reason.
3. If the job is not running, you can restart it ( alternatively you can use jenkins job to deploy the corresponding job, if you don't have access to restart).

Hope this helps!
cc: @aks30 @vijiurs
